src/main/webapp/add_event_page/script.js

```diff
@@ -1,22 +1,122 @@
-        position: {lat:32.8811, lng:-117.2376},
+        position: {lat:initial_lat, lng:initial_lng},
         map: map,
         draggable: true,
     });
 
+    const input = this.document.getElementById("place-input");
+    const searchBox = new google.maps.places.SearchBox(input);
+
+    google.maps.event.addListener(searchBox,'places_changed',function(){
+        var places = searchBox.getPlaces();
+        var bounds = new google.maps.LatLngBounds();
+        var i,place;
+        for(i=0; place=places[i]; i++){
+          bounds.extend(place.geometry.location);
+          marker.setPosition(place.geometry.location);
+        }
+        map.fitBounds(bounds);
+        map.setZoom(15);
+        currentLatitude = marker.getPosition().lat();
+        currentLongitude = marker.getPosition().lng();
+    });
+
     google.maps.event.addListener(marker, 'dragend', function(marker){
         var latLng = marker.latLng; 
         currentLatitude = latLng.lat();
         currentLongitude = latLng.lng();
-        jQ("#latitude").val(currentLatitude);
-        jQ("#longitude").val(currentLongitude);
     }); 
 }
-window.initMap = initMap;
\ No newline at end of file
+window.initMap = initMap;
+
+function checkType(){
+    if (this.document.getElementById("event_type").value == "Study") {
+        this.document.getElementById("subject1").style.display = '';
+        this.document.getElementById("subject2").style.display = '';
+    } else {
+        this.document.getElementById("subject1").style.display = 'none';
+        this.document.getElementById("subject2").style.display = 'none';
+    }
+}
+
+function checkMin(){
+    now = new Date();
+    formatted_date = now.getFullYear() + '-' +
+        String(now.getMonth() + 1).padStart(2, '0') + '-' +
+        String(now.getDate()).padStart(2, '0') + 'T' +
+        String(now.getHours()).padStart(2, '0') + ':' +
+        String(now.getMinutes()).padStart(2, '0');
+    this.document.getElementById('date').setAttribute('min', formatted_date);
+}
+
+function submitEvent(){
+    var event_name = document.getElementById("name").value;
+    var event_des = document.getElementById("description").value;
+    var event_loc = document.getElementById("location_name").value;
+    var event_date = new Date(document.getElementById("date").value).toISOString();
+    var event_type = document.getElementById("event_type").value;
+    if (event_type == "Social"){
+        var event_sub = "";
+    }
+    else{
+        var event_sub = document.getElementById("subject").value;
+    }
+    var data = { 
+        "name": event_name,
+        "description": event_des,
+        "loc": event_loc,
+        "date": event_date,
+        "type": event_type,
+        "subject": event_sub,
+        "lat": currentLatitude,
+        "lng": currentLongitude,
+        "school_id": school_id,
+    }
+    
+    fetch("/event",{
+         method: "POST",
+         body: JSON.stringify(data),
+    });
+}
```

Comment:
At the end of the submitEvent() function this line would be added.
``window.location.href = `../directory-page/directory.html?school_id=${school_id}`;``